### PR TITLE
make LB loss with sigmoid more robust

### PR DESCRIPTION
**Pull request by epwalsh**
The epsilon we used in the denominator when normalizing the sigmoid scores for the LB loss was too big (1e-20, which we took from [Megatron-LM](https://github.com/NVIDIA/Megatron-LM/blob/55c968dabb6d32cf8c21b1e3fadeea47e2ceb78d/megatron/core/transformer/moe/router.py#L195-L199)), allowing the model to hack the LB loss function by pushing sigmoid scores below epsilon.

image

### CODE REVIEW

File: src/olmo_core/nn/moe/router.py [hunk 1]
```diff
@@ -492,7 +494,7 @@ def forward(
             if self.z_loss_weight is not None:
                 assert self.z_loss is not None
                 z_loss = router_z_loss(
-                    expert_logits=logits.float(),
```
Comment (epwalsh) [inline src/olmo_core/nn/moe/router.py:495]: This was redundant.

File: src/olmo_core/nn/moe/router.py [hunk 2]
```diff
@@ -455,10 +458,9 @@ def forward(
                 )
             )
 
-        scores = scores.float()
         # Make sure scores are normalized, otherwise load balancing loss doesn't work.
         if self.gating_function == MoERouterGatingFunction.sigmoid:
-            scores = scores / (scores.sum(dim=-1, keepdim=True) + 1e-20)
+            scores = scores / (scores.sum(dim=-1, keepdim=True) + _EPSILON)
```
Comment (2015aroras) [inline src/olmo_core/nn/moe/router.py]: Is the issue that the epsilon was too big?

Comment (epwalsh) [reply src/olmo_core/nn/moe/router.py]: Yeup, surprisingly
